resource/resource_test.go

```diff
@@ -309,6 +309,62 @@ func TestOutE2E(t *testing.T) {
 	}
 }
 
+func TestTargetURL(t *testing.T) {
+	testCases := map[string]struct {
+		atc          string
+		team         string
+		pipeline     string
+		job          string
+		buildN       string
+		instanceVars string
+		want         string
+	}{
+		"all defaults": {
+			want: "https://ci.example.com/teams/devs/pipelines/magritte/jobs/paint/builds/42",
+		},
+		"instanced vars 1": {
+			instanceVars: `{"branch":"stable"}`,
+			want:         "https://ci.example.com/teams/devs/pipelines/magritte/jobs/paint/builds/42?vars=%7B%22branch%22%3A%22stable%22%7D",
+		},
+		"instanced vars 2": {
+			instanceVars: `{"branch":"stable","foo":"bar"}`,
+			want:         "https://ci.example.com/teams/devs/pipelines/magritte/jobs/paint/builds/42?vars=%7B%22branch%22%3A%22stable%22%2C%22foo%22%3A%22bar%22%7D",
```

Comment:
For my understanding: what is the meaning of the second variable? How many pipelines we have in this case two (`stable` and `bar`) or three (`master`, `stable` and `bar`)?

Reply:
Suppose you want to set an instanced pipeline for each PR to a repo, and you want to test each PR against multiple Go versions. You may have pipelines with multiple instance vars in that case: `{"number": 123, "go": "1.17"}`. Or maybe you want to group all PRs across a set of repos, in which case you might have `{"repo": "cogito", "number": 123}`.

In general, you may want an N-dimensional "matrix" of instanced pipelines